File: public/docs/_examples/dependency-injection/ts/app/user.service.ts
// #docregion
import {Injectable} from 'angular2/core';

@Injectable()
export class UserService {
  // Todo: get the user; don't 'new' it.
  private _alice = new User('Alice', true);
  private _bob = new User('Bob', false);

  // initial user is Bob
  user = this._bob;

  // swaps users
  getNewUser() {
    return this.user = this.user === this._bob ? this._alice : this._bob;
  }
}

export class User {
  constructor(
    public name:string,
    public isAuthorized:boolean = false) { }
}

File: public/docs/_examples/dependency-injection/dart/lib/user_service.dart
// #docregion
import 'package:angular2/core.dart';

class User {
  final String name;
  final bool isAuthorized;

  User(this.name, [this.isAuthorized = false]);
}

// Todo: get the user; don't 'new' it.
final User _alice = new User('Alice', true);
final User _bob = new User('Bob', false);

@Injectable()
class UserService {
  User user;

  UserService() : user = _bob;

  // swap users
  User getNewUser() => user = user == _bob ? _alice : _bob;
}
